spec/amber/controller/base_spec.cr

```diff
@@ -83,6 +83,44 @@ module Amber::Controller
         context.request.headers["Accept"] = "text/plain"
         ResponsesController.new(context).index.should eq expected_result
       end
+
+      it "responds with json for path.json" do
+        expected_result = %({"type":"json","name":"Amberator"})
+        context.request.path = "/response/1.json"
+        ResponsesController.new(context).index.should eq expected_result
+      end
+
+      it "responds with xml for path.xml" do
+        expected_result = "<xml><body><h1>Sort of xml</h1></body></xml>"
+        context.request.path = "/response/1.xml"
+        ResponsesController.new(context).index.should eq expected_result
+      end
+
+      it "responds with text for path.txt" do
+        expected_result = "Hello I'm text!"
+        context.request.path = "/response/1.txt"
+        ResponsesController.new(context).index.should eq expected_result
+      end
+
+      it "responds with text for path.text" do
+        expected_result = "Hello I'm text!"
+        context.request.path = "/response/1.text"
+        ResponsesController.new(context).index.should eq expected_result
+      end
+
+      it "responds with 406 for path.text when text hasn't been defined" do
+        expected_result = "Response unexceptable."
+        context.request.path = "/response/1.text"
+        ResponsesController.new(context).show.should eq expected_result
+        context.response.status_code.should eq 406
+      end
+
+      it "respond_with default or accept header request if extension is invalid" do
+        expected_result = "<html><body><h1>Elorest <3 Amber</h1></body></html>"
+        context.request.path = "/response/1.texas"
+        context.request.headers["Accept"] = "text/html"
+        ResponsesController.new(context).index.should eq expected_result
+      end
```

Comment:
I think so. I'll check it I guess. Since they're all set at the some time I didn't think it was important to set that since the response was right.

Reply:
For some reason I'm getting an error that `content_type` doesn't exist on response. When I try to test it's value.

Reply:
I can set it though.

Reply:
In base_spec we should be verifying the context `context.response` as well. This is `ResponsesController.new(context).index.should eq expected_result` totally valid, at least confirm the content-type and status code are correct. For instance 

```crystal
context.response.body.should
context.response.status_code.should
context.response.headers["Content-Type"].should
```
I think it is important that is done this way because something else could be changing the response. This is how I found the issue with the `redirect` :)

Reply:
Unless you are covering testing status code and content-type in the unit test for Responders::Content context.response.status_code.should
context.response.headers["Content-Type"].should

Reply:
Good point. I was put off by the fact that there was no getter for content_type. Makes sense that it's getting written to the header though. I'll test that.

Reply:
`context.response.headers["Content-Type"]?` will report a failure with helpful info rather than an error when `nil`